crates/sqlexec/src/planner/session_planner.rs

```diff
@@ -947,6 +953,61 @@ impl<'a> SessionPlanner<'a> {
                 Ok(VariablePlan::ShowVariable(ShowVariable { variable }).into())
             }
 
+            // "DELETE FROM <table> WHERE <expression>"
+            //
+            // deletes rows from a table that matches the expression.
+            // or all the rows if no expression is provided.
+            ast::Statement::Delete {
+                tables,
+                from,
+                using: None,
+                selection,
+                returning: None,
+            } if tables.is_empty() => {
+                let (table_name, schema) = match from.len() {
+                    0 => {
+                        return Err(PlanError::InvalidDeleteStatement {
+                            msg: "DELETE FROM should have atleast one table name",
+                        })
+                    }
+                    1 => {
+                        let table_factor = from[0].relation.clone();
```

Comment:
We can avoid this clone here easily. Since there's just one element in the array, and it's not used again, you can:

```suggestion
                        let table_factor = from.pop().relation;
```

Reply:
from is immutable here

Reply:
You can add `mut` in front of `from`. It's owned here so you should be able to do anything with it:

```rust
            ast::Statement::Delete {
                 // ...
                 mut from,
                 // ...
             } if tables.is_empty() => {
```